### DISCUSSION TITLE
Is there a way of inferring a type based on some dynamic input?

### DISCUSSION THREAD

**Discussion by pedropmedina** [trpc/trpc#3626] [General]
Hi,

I'm generating zod types based on schema.prisma file for use in trpc routers. This allows more flexibility as to how we `include` or `select` data from prisma since I can tell the trpc client -> `trpcAPI.contact.findMany({ include:  { address: true }}). 

However when using the trpc helper to infer the types `inferRouterOutputs['contact']['findMany']` I don't get back the `contact.address`. 

I wouldn't want to reference the `address` directly in that route as I only want to call it when I need it, yet I do need the type for some of the components in the frontend.

Is there a way of using `inferRouterOutputs['contact']['findMany']` and get the `address` reference type without directly defining it in the trpc route?

**Comment by bshearrer** [comment]
HI @pedropmedina, have you found a good way to handle this since posting?

**Comment by KATT** [comment]
See #2150
